examples/assets_dynamic_partitions/assets_dynamic_partitions/assets.py

```diff
@@ -0,0 +1,125 @@
+import json
+import os
+import urllib.request
+import zipfile
+
+import pendulum
+import requests
+from dagster import DynamicPartitionsDefinition, Output, asset, get_dagster_logger
+from pandas import DataFrame
+
+releases_partitions_def = DynamicPartitionsDefinition(name="releases")
+
+
+@asset(
+    partitions_def=releases_partitions_def,
+    io_manager_key="warehouse",
+    metadata={"partition_expr": "release_tag"},
+)
+def releases_metadata(context) -> DataFrame:
+    """Table with a single row per release"""
+    release_tag = context.partition_key
```

Comment:
Throughout... should we use `context.asset_partition_key_for_output()` or `context.partition_key`? I prefer the later but the docs use the former: https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#defining-partitioned-assets

Reply:
Hmm good question. The former is theoretically more principled, but I'm just going to switch the docs to the latter.